apps/api/src/app/notification-template/usecases/create-blueprint-notification-template/create-blueprint-notification-template.usecase.ts

```diff
@@ -0,0 +1,132 @@
+      CreateNotificationTemplateCommand.create({
+        organizationId: command.organizationId,
+        userId: command.userId,
+        environmentId: command.environmentId,
+        name: template.name,
+        tags: template.tags,
+        description: template.description,
+        steps: steps,
+        notificationGroupId: group._id,
+        active: template.active ?? false,
+        draft: template.draft ?? true,
+        critical: template.critical ?? false,
+        preferenceSettings: template.preferenceSettings,
+        blueprintId: command.templateId,
+      })
+    );
+  }
+
+  private async handleFeeds(
+    steps: NotificationStepEntity[],
+    command: CreateBlueprintNotificationTemplateCommand
+  ): Promise<NotificationStepEntity[]> {
+    for (let i = 0; i < steps.length; i++) {
+      const step = steps[i];
+
+      if (!step.template._feedId) {
+        continue;
+      }
+
+      const originalFeed = await this.feedRepository.findOne({
+        _organizationId: this.getBlueprintOrganizationId(),
+        id: step.template._feedId,
+      });
+
+      if (!originalFeed) {
+        step.template._feedId = undefined;
+        steps[i] = step;
+        continue;
+      }
+
+      let foundFeed = await this.feedRepository.findOne({
+        _organizationId: command.organizationId,
+        identifier: originalFeed.identifier,
+      });
+
+      if (!foundFeed) {
+        foundFeed = await this.feedRepository.create({
+          name: foundFeed.name,
+          identifier: foundFeed.identifier,
+          _environmentId: command.environmentId,
+          _organizationId: command.organizationId,
+        });
+      }
+
+      step.template._feedId = foundFeed._id;
+      steps[i] = step;
+    }
+
+    return steps;
+  }
+
+  private async handleGroup(
+    notificationGroupId: string,
+    command: CreateBlueprintNotificationTemplateCommand
+  ): Promise<NotificationGroupEntity> {
+    let group = await this.notificationGroupRepository.findOne({
+      name: 'General',
+      _organizationId: command.organizationId,
+    });
+
+    if (!group) {
+      throw new NotFoundException(`Notification group with name General not found`);
+    }
+
+    const originalGroup = await this.notificationGroupRepository.findOne({
+      _id: notificationGroupId,
+      _organizationId: this.getBlueprintOrganizationId(),
+    });
+
+    if (originalGroup && originalGroup.name !== group.name) {
+      group = await this.notificationGroupRepository.findOne({
+        name: originalGroup.name,
+        _organizationId: command.organizationId,
+      });
+    }
+
+    if (!group) {
+      group = await this.notificationGroupRepository.create({
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        name: originalGroup.name,
+      });
+    }
+
+    return group;
+  }
+
+  private getBlueprintOrganizationId(): string {
+    return process.env.BLUEPRINT_CREATOR;
+  }
```

Comment:
This is a good thing even if it is a small function. There are more places where we need to bring the env variable BLUEPRINT_CREATOR. We can think of having a helper function to reuse everywhere. Maybe I am biased that it took me a while and many questions to understand this feature.